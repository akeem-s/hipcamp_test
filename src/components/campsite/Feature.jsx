import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

const propTypes = {
  presence: PropTypes.bool,
  subfeatures: PropTypes.array,
  title: PropTypes.string,
};

export default class Feature extends Component {
  constructor(props){
    super(props);
    this.renderPresenceIcons = this.renderPresenceIcons.bind(this);
    this.handleClick = this.handleClick.bind(this);
    state:{
      showSubfeatures: false;
    }
  }

  renderPresenceIcons(presence){
    if(presence){
      return <FontAwesome className='customFaCheck customFa' name='check'/>;
    }
    return <FontAwesome className='customFaBan customFa' name='ban'/>;
  };

  handleClick(subfeatures){

  };

  renderSubfeatures(){
    const { subfeatures } = this.props;
    if(subfeatures.length > 0){
      subfeatures.map((f)=>{
        return (
          <div className='subfeaturesContainer' key={f.title}>
            <p className='subfeatureTitle'>{f.title}</p>
            <p className='subfeaturePresence'>{f.presence}</p>
          </div>
        )
      })
    }
  };

  render(){
    const { presence, subfeatures, title } = this.props;
    return (
      <div className='feature' >
        <p className='featureTitle' onClick={()=>{this.handleClick(subfeatures)}}>{title}</p>
        {this.}
        {this.renderPresenceIcons(presence)}
      </div>
    )
  }
};

Feature.propTypes = propTypes;
