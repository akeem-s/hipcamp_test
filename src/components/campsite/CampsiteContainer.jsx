import React, { Component } from 'react';
// components
import Feature from './Feature';
import { FEATURES } from '../../assets/camp_features';

export default class CampsiteContainer extends Component {
  constructor(props){
    super(props);
    this.state = {
      features:[],
    };
  }

  componentDidMount(){
    // Note: I am setting state here because I am treating this data like an api response
    // and the React docs suggest instantiating API calls in componentDidMount. Setting state
    // here can have performance implications, which I would consider if dealing with real
    // data. The state of this container would normally manage the state of the
    // api call (isLoading, hasError, etc)
    this.setState({
      features: FEATURES,
    });
  }

  render(){
    return (
      <div className='campsiteContainer'>
        <h2 className='campsiteName'>Campsite Name Placeholder</h2>
        {this.state.features.map((f)=>{ return <Feature key={f.title} title={f.title} presence={f.presence} subfeatures={f.subfeatures}/>; })}
      </div>
    );
  }
}
