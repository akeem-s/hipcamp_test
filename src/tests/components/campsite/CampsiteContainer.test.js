import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CampsiteContainer from '../../../components/campsite/CampsiteContainer';
import Feature from '../../../components/campsite/Feature';
import { FEATURES } from '../../../assets/camp_features.js';

describe('CampsiteContainer', () => {
  let wrapper;
  const getWrapper = () => {
    const component = shallow(<CampsiteContainer/>);
    return component;
  };

  beforeEach(() => {
    wrapper = getWrapper();
  });

  test('component renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('includes Feature', () => {
    expect(wrapper.find(Feature).length).toEqual(FEATURES.length);
  });

  test('snapshot should contain expected DOM elements', () => {
    const renderedValue = renderer.create(wrapper).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
