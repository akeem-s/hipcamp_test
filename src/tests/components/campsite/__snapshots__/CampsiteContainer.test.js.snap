// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampsiteContainer snapshot should contain expected DOM elements 1`] = `
<div
  className="campsiteContainer"
>
  <h2
    className="campsiteName"
  >
    Campsite Name Placeholder
  </h2>
  <div
    className="feature null"
    onClick={[Function]}
  >
    <div
      className="featureDetailsContainer"
    >
      <p
        className="featureTitle"
      >
        Toilet
      </p>
      <span
        aria-hidden={true}
        className="fa fa-check customFaCheck customFa"
      />
    </div>
  </div>
  <div
    className="feature null"
    onClick={[Function]}
  >
    <div
      className="featureDetailsContainer"
    >
      <p
        className="featureTitle"
      >
        Pets allowed
      </p>
      <span
        aria-hidden={true}
        className="fa fa-check customFaCheck customFa"
      />
    </div>
  </div>
  <div
    className="feature subfeaturesPresent"
    onClick={[Function]}
  >
    <div
      className="featureDetailsContainer"
    >
      <p
        className="featureTitle"
      >
        Shower
      </p>
      <span
        aria-hidden={true}
        className="fa fa-ban customFaBan customFa"
      />
    </div>
  </div>
  <div
    className="feature subfeaturesPresent"
    onClick={[Function]}
  >
    <div
      className="featureDetailsContainer"
    >
      <p
        className="featureTitle"
      >
        Trash
      </p>
      <span
        aria-hidden={true}
        className="fa fa-check customFaCheck customFa"
      />
    </div>
  </div>
</div>
`;
